checker/src/summaries.rs

```diff
@@ -41,12 +41,11 @@ use std::ops::Deref;
 ///    desirable to havoc all static variables every time such a function is called. Consequently
 ///    sound analysis is only possible one can assume that all such functions have been provided
 ///    with explicit contract functions.
-#[derive(Serialize, Deserialize, Clone, Debug, Default, Eq)]
+#[derive(Serialize, Deserialize, Clone, Debug, Default, Eq, PartialEq)]
 pub struct Summary {
-    /// The number of seconds that the analyzer used to construct this summary.
-    /// If it is >= k_limits::MAX_ANALYSIS_TIME_FOR_BODY then this summary is not definitive
-    /// because the analyzer will omit it from the outer fixed point loop.
-    pub analysis_time_in_seconds: u64,
+    /// If true this summary was computed. If false, it is a default summary.
+    /// Used to distinguish a computed empty summary from a default summary.
+    pub is_not_default: bool,
```

Comment:
I find 'is_default' a little more intuitive, otherwise you end up with '!is_not_default' and whatnot. But I guess if you manually avoid logic like that then it's ok. I'm guessing you did this because then a Summary::default() and whatnot will have the right value.